Add recipe for consult-hn

agzam opened the pull request:
### Brief summary of what the package does

Consult extension for searching on HackerNews

### Direct link to the package repository

https://github.com/agzam/consult-hn

### Your association with the package

Author

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [X] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [X] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [X] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [X] My elisp byte-compiles cleanly
- [X] I've used `M-x checkdoc` to check the package's documentation strings
- [X] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.   Your package `;;; Commentary` says:
```elisp
;;; Commentary:
;;
;;  Description
;;
;;  This is an extension for Consult
;;  https://github.com/minad/consult
```
The additional "Description" header is unnecessary, but I would definitely go into more detail about what this extension does.

⸺ `consult-hn.el` with byte-compile using Emacs 30.1:
```
consult-hn.el:271:10: Warning: Unused lexical variable `nbPages'
```
(Also consider that `nbPages` is written in camelcase, which is nonstandard in Elisp)

⸺ `consult-hn-transient.el` with package-lint 20250527.1845 and `package-lint-main-file` = "consult-hn.el":
```
1 issue found:
11:0: error: Package-Requires outside the main file have no effect.
```

⸺ Package and license:
- consult-hn.el must include all of the Package-Requires listed in consult-hn-transient.el, including: transient "0.9"
- consult-hn-transient.el needs *formal* license boilerplate and/or an [SPDX-License-Identifier](https://spdx.dev/ids/)
- consult-hn.el needs *formal* license boilerplate and/or an [SPDX-License-Identifier](https://spdx.dev/ids/)

agzam commented:
Thank you Chris, for your generosity of time and beautiful job of reviewing my submission. I have addressed your comments. It seems I keep missing the SPDX-License-Identifier and keep forgetting to remove ';;Description'. I need to update my template. It would be also nice if package-lint said something about the missing header; maybe I'll try to find out about that.

Let me know if I missed anything else. Cheers!

riscy commented:
Thanks for the update!  Here's the latest -

⸺ `consult-hn.el` with byte-compile using Emacs 30.1:
```
consult-hn.el:279:23: Warning: assignment to free variable `nb-pages'
```

⸺ `consult-hn-transient.el` with package-lint 20250527.1845 and `package-lint-main-file` = "consult-hn.el":
```
1 issue found:
11:0: error: Package-Requires outside the main file have no effect.
```

agzam commented:
I apologize for the delay — I'm on vacation at the moment. I couldn't reproduce the issues you listed - neither the var thing, nor the byte compilation. There's a caveat though - I currently have only my MBP with me and I couldn't try it on Linux. If you get a chance to check things once more before next week and still see the issues, would you mind telling me what I could try to reproduce them? In any case, I will be back home next week and will try it on Linux. At least on Mac, things look okay on my side.
